bundles/org.openhab.binding.flicbutton/src/main/java/org/openhab/binding/flicbutton/handler/FlicDaemonBridgeHandler.java

```diff
@@ -0,0 +1,150 @@
+import org.slf4j.LoggerFactory;
+
+import io.flic.fliclib.javaclient.FlicClient;
+
+/**
+ * The {@link FlicDaemonBridgeHandler} handles a running instance of the fliclib-linux-hci server (flicd).
+ *
+ * @author Patrick Fink - Initial contribution
+ */
+public class FlicDaemonBridgeHandler extends BaseBridgeHandler {
+    private static final Logger logger = LoggerFactory.getLogger(FlicDaemonBridgeHandler.class);
+    private static final long REINITIALIZE_DELAY_SECONDS = 10;
+    // Config parameters
+    private FlicDaemonBridgeConfiguration cfg;
+    // Services
+    private FlicButtonDiscoveryService buttonDiscoveryService;
+    private Future<?> flicClientFuture;
+    // For disposal
+    private Collection<Future<?>> startedTasks = new ArrayList<>(2);
+    private FlicClient flicClient;
+
+    public FlicDaemonBridgeHandler(Bridge bridge, FlicButtonDiscoveryService buttonDiscoveryService) {
+        super(bridge);
+        this.buttonDiscoveryService = buttonDiscoveryService;
+    }
+
+    public FlicClient getFlicClient() {
+        return flicClient;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initialize Fliclib bridge");
+
+        try {
+            initConfigParameters();
+            startFlicdClientAsync();
+            activateButtonDiscoveryService();
+            initThingStatus();
+        } catch (UnknownHostException ignored) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Hostname wrong or unknown!");
+        } catch (IllegalArgumentException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration (hostname, port) is invalid and cannot be parsed.");
+        } catch (IOException e) {
+            logger.warn("Error occured while connecting to flicd", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error connecting to flicd!");
+        }
+    }
+
+    private void initConfigParameters() throws UnknownHostException {
+        Object hostConfigRaw = thing.getConfiguration().get(FlicButtonBindingConstants.CONFIG_HOST_NAME);
+        Object portConfigRaw = thing.getConfiguration().get(FlicButtonBindingConstants.CONFIG_PORT);
+        cfg = new FlicDaemonBridgeConfiguration(hostConfigRaw, portConfigRaw);
+    }
+
+    private void activateButtonDiscoveryService() {
+        buttonDiscoveryService.activate(flicClient);
+    }
+
+    private void startFlicdClientAsync() throws IOException {
+        flicClient = new FlicClient(cfg.getHostname().getHostAddress(), cfg.getPort());
+        Runnable flicClientService = () -> {
+            try {
+                flicClient.handleEvents();
+                logger.info("Listening to flicd unexpectedly ended");
+            } catch (Exception e) {
+                logger.info("Error occured while listening to flicd", e);
+            } finally {
+                onClientFailure();
```

Comment:
This will also be invoked on success. Is this intentional?

Reply:
handleEvents() is an infinite loop that should never reach it's end. So if this line is reached, it's always a failure.